src/controllers/auth.controller.ts

```diff
@@ -84,6 +85,98 @@ export const refreshToken = async (
   }
 };
 
+const getOrCreateUser = async (
```

Comment:
are we doing this also for github?  

i believe originally i didn't want to create users unless absolutely necessary, but i guess can can, we might want later to create custodial dids for them

should we refactor out the create user part and just make the differences for linkedin and github?

Reply:
yes we're also doing this for github. matter of fact, i extracted this part from authenticateGithub and made it a separate function.